dhis-2/dhis-api/src/main/java/org/hisp/dhis/dataapproval/DataApprovalAudit.java

```diff
@@ -275,114 +277,21 @@ public boolean equals( Object object )
         {
             return true;
         }
-
-        if ( object == null || !(object instanceof DataApprovalAudit) )
-        {
-            return false;
-        }
-
-        DataApprovalAudit that = (DataApprovalAudit) object;
-
-        if ( id != that.id )
-        {
-            return false;
-        }
-        else if ( level != null )
-        {
-            if ( !level.equals( that.getLevel() ) )
-            {
-                return false;
-            }
-        }
-        else if ( that.getLevel() != null )
-        {
-            return false;
-        }
-
-        if ( workflow != null )
-        {
-            if ( !workflow.equals( that.getWorkflow() ) )
-            {
-                return false;
-            }
-        }
-        else if ( that.getWorkflow() != null )
-        {
-            return false;
-        }
-
-        if ( period != null )
-        {
-            if ( !period.equals( that.getPeriod() ) )
-            {
-                return false;
-            }
-        }
-        else if ( that.getPeriod() != null )
-        {
-            return false;
-        }
-
-        if ( organisationUnit != null )
-        {
-            if ( !organisationUnit.equals( that.getOrganisationUnit() ) )
-            {
-                return false;
-            }
-        }
-        else if ( that.getOrganisationUnit() != null )
-        {
-            return false;
-        }
-
-        if ( attributeOptionCombo != null )
-        {
-            if ( !attributeOptionCombo.equals( that.getAttributeOptionCombo() ) )
-            {
-                return false;
-            }
-        }
-        else if ( that.getAttributeOptionCombo() != null )
-        {
-            return false;
-        }
-
-        if ( action != null )
-        {
-            if ( !action.equals( that.getAction() ) )
-            {
-                return false;
-            }
-        }
-        else if ( that.getAction() != null )
-        {
-            return false;
-        }
-
-        if ( created != null )
-        {
-            if ( !created.equals( that.getCreated() ) )
-            {
-                return false;
-            }
-        }
-        else if ( that.getCreated() != null )
-        {
-            return false;
-        }
-
-        if ( creator != null )
-        {
-            if ( !creator.equals( that.getCreator() ) )
-            {
-                return false;
-            }
-        }
-        else if ( that.getCreator() != null )
+        if ( !(object instanceof DataApprovalAudit) )
         {
             return false;
         }
 
```

Comment:
Looks good. Should we use Objects.hash or Objects.hashCode for hashCode just to make it consistent and equally modern while at it?

Reply:
We absolutely should. Done.